### ISSUE TITLE
Using minWidth: double.maxFinite to expand does not work properly if the ToggleSwitch is inside Padding or other widget

### ISSUE THREAD

**Issue by poqueque** [PramodJoshi/toggle_switch#79]
Using minWidth: double.maxFinite to expand does not work properly if the ToggleSwitch is inside Padding or other widget

```
Padding(
            padding: const EdgeInsets.all(8.0),
            child: ToggleSwitch(
              minWidth: double.maxFinite,
              activeBgColors: [
                [AppColors.darkBlue100],
                [AppColors.darkBlue100],
              ],
              activeFgColor: Colors.white,
              inactiveBgColor: AppColors.disabledTile,
              inactiveFgColor: Colors.black,
              initialLabelIndex: 1,
              totalSwitches: 2,
              labels: ['Manual', 'Auto'],
              radiusStyle: true,
              onToggle: (index) {
                print('switched to: $index');
              },
            ),
          )
```

![Captura de Pantalla 2023-07-05 a las 12 43 45](https://github.com/PramodJoshi/toggle_switch/assets/902418/feaced92-3fe3-47d5-a90d-fcf4f9759b93)

**Comment by PramodJoshi** [comment]
Hey, I was unable to re-create this issue with the latest version,

```dart
import 'package:flutter/material.dart';
import 'package:toggle_switch/toggle_switch.dart';

void main() => runApp(
  MaterialApp(
    home: MyApp(),
  ),
);

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Center(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: ToggleSwitch(
                minWidth: double.maxFinite,
                activeBgColors: [
                  [Colors.blue],
                  [Colors.blue],
                ],
                activeFgColor: Colors.white,
                inactiveBgColor: Colors.grey,
                inactiveFgColor: Colors.black,
                initialLabelIndex: 1,
                totalSwitches: 2,
                labels: ['Manual', 'Auto'],
                radiusStyle: true,
                onToggle: (index) {
                  print('switched to: $index');
                },
              ),
            ),
        ),
      ),
    );
  }
}
```

https://github.com/PramodJoshi/toggle_switch/assets/17320471/972372cd-ad6a-4e81-bd09-a0e17a81dd2b

**Comment by namhq1989** [comment]
Same problem here. It's ok if I skip option `minWidth`

![image](https://github.com/PramodJoshi/toggle_switch/assets/1388374/de4e20a4-e67c-4401-b1d9-1cabd3a10b6a)

- Version: ^2.1.0
- Code:

```dart
return Padding(
    padding: const EdgeInsets.all(20.0),
    child: ToggleSwitch(
      minWidth: double.maxFinite,
      minHeight: 32,
      cornerRadius: 24,
      activeBgColor: const [Colors.white],
      activeFgColor: Theme.of(context).colorScheme.primary,
      inactiveBgColor: const Color.fromRGBO(164, 39, 81, 1),
      inactiveFgColor: Colors.white,
      initialLabelIndex: 0,
      totalSwitches: 2,
      labels: const ['Month', 'Year'],
      radiusStyle: true,
      onToggle: (index) {
        AppLogger.print('switched to: $index');
      },
    ),
);
```

**Comment by namhq1989** [comment]
@PramodJoshi still got overflow with your code. Here is screenshot and my code:

![image](https://github.com/PramodJoshi/toggle_switch/assets/1388374/a3911710-d68c-44c7-bf00-263485437850)

```dart
import 'package:auto_route/auto_route.dart';
import 'package:flutter/material.dart';
import 'package:toggle_switch/toggle_switch.dart';

@RoutePage()
class GameScreen extends StatelessWidget {
  const GameScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Game'),
      ),
      body: Center(
        // child: Text('GAME SCREEN'),
        child: _buildRanking(context: context),
      ),
    );
  }
}

Widget _buildRanking({required BuildContext context}) {
  return Container(
    height: 300,
    decoration: const BoxDecoration(
      color: Colors.black,
    ),
    child: Container(
      decoration: const BoxDecoration(
        color: Color.fromRGBO(205, 51, 103, 1),
      ),
      padding: const EdgeInsets.all(20.0),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          const Text(
            'Ranking',
            style: TextStyle(
              fontSize: 20,
              color: Colors.white,
              fontWeight: FontWeight.w700,
            ),
          ),
          const SizedBox(height: 4.0),
          const Text(
            'Lorem ipsum',
            style: TextStyle(
              fontSize: 14,
              color: Colors.white,
            ),
          ),
          const SizedBox(height: 20.0),
          Center(
            child: ToggleSwitch(
              minWidth: double.maxFinite,
              activeBgColors: const [
                [Colors.blue],
                [Colors.blue],
              ],
              activeFgColor: Colors.white,
              inactiveBgColor: Colors.grey,
              inactiveFgColor: Colors.black,
              initialLabelIndex: 1,
              totalSwitches: 2,
              labels: const ['Manual', 'Auto'],
              radiusStyle: true,
              onToggle: (index) {
                print('switched to: $index');
              },
            ),
          ),
        ],
      ),
    ),
  );
}
```

**Comment by namhq1989** [comment]
hi @PramodJoshi , any news on this?

**Comment by PramodJoshi** [comment]
Hi @namhq1989, sorry, haven't had a chance to test your code. I'll try to do that by this weekend and let you know.

**Comment by PramodJoshi** [comment]
Hi @namhq1989, sorry for the delay. I was able to re-create this issue. Seems like it's caused by extra padding. I'll work on a fix asap.

In the meantime, you can remove the padding for ToggleSwitch and just wrap other children with Column and Padding,

```dart
import 'package:flutter/material.dart';
import 'package:toggle_switch/toggle_switch.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // TRY THIS: Try running your application with "flutter run". You'll see
        // the application has a blue toolbar. Then, without quitting the app,
        // try changing the seedColor in the colorScheme below to Colors.green
        // and then invoke "hot reload" (save your changes or press the "hot
        // reload" button in a Flutter-supported IDE, or press "r" if you used
        // the command line to start the app).
        //
        // Notice that the counter didn't reset back to zero; the application
        // state is not lost during the reload. To reset the state, use hot
        // restart instead.
        //
        // This works for code too, not just values: Most code changes can be
        // tested with just a hot reload.
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Game'),
      ),
      body: Center(
        // child: Text('GAME SCREEN'),
        child: _buildRanking(context: context),
      ),
    );
  }
}

Widget _buildRanking({required BuildContext context}) {
  return Container(
    height: 300,
    decoration: const BoxDecoration(
      color: Colors.black,
    ),
    child: Container(
      decoration: const BoxDecoration(
        color: Color.fromRGBO(205, 51, 103, 1),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Padding(
            padding: const EdgeInsets.all(20.0),
            child: Column(
              children: [
                const Text(
                  'Ranking',
                  style: TextStyle(
                    fontSize: 20,
                    color: Colors.white,
                    fontWeight: FontWeight.w700,
                  ),
                ),
                const SizedBox(height: 4.0),
                const Text(
                  'Lorem ipsum',
                  style: TextStyle(
                    fontSize: 14,
                    color: Colors.white,
                  ),
                ),
                const SizedBox(height: 20.0),
              ],
            ),
          ),
          Center(
            child: ToggleSwitch(
              minWidth: double.maxFinite,
              activeBgColors: const [
                [Colors.blue],
                [Colors.blue],
              ],
              activeFgColor: Colors.white,
              inactiveBgColor: Colors.grey,
              inactiveFgColor: Colors.black,
              initialLabelIndex: 1,
              totalSwitches: 2,
              labels: const ['Manual', 'Auto'],
              radiusStyle: true,
              onToggle: (index) {
                print('switched to: $index');
              },
            ),
          ),
        ],
      ),
    ),
  );
}
```

<img width="418" alt="Screenshot 2023-08-08 at 4 41 57 PM" src="https://github.com/PramodJoshi/toggle_switch/assets/17320471/f31ce2dc-2cac-4ff4-9d31-eaf8a72bc7cd">

**Comment by namhq1989** [comment]
Thank you @PramodJoshi , I'll change my code and waiting for the next patch.

**Comment by langhoangal** [comment]
The same here, but remove the top padding then adding it to each child takes much time in my case. So, using layout builder is a better solution. 
`minWidth: constraints.biggest.width / numOfTabs - 1`

**Comment by AhmedLSayed9** [comment]
Can you re-open this issue?

**Comment by PramodJoshi** [comment]
This has been fixed in the latest version. Cheers!

**Comment by derklaro** [comment]
Hi @PramodJoshi,

I am not sure what exactly you did to fix this issue, however there seems to be something wrong with it: starting from version 2.2.2 I experience a weird offset of my labels, which breaks my pre-calculated custom widths for them, but they also doesn't appear centered anymore.

2.2.1:
![image](https://github.com/PramodJoshi/toggle_switch/assets/40468651/02890650-8fa9-48a1-8b21-3f805787febc)

2.2.2:
![image](https://github.com/PramodJoshi/toggle_switch/assets/40468651/2c952351-4f69-4cae-97b2-015740d852cd)

This is my current code, which should be useable to reproduce the issue (if not let me know so I can give more context):
```dart
  Row(
    children: [
      const Text(
        "Filter:",
        style: TextStyle(
          fontSize: 15,
          fontWeight: FontWeight.bold,
        ),
      ),
      const SizedBox(
        width: 5,
      ),
      ToggleSwitch(
        fontSize: 15,
        minHeight: 25,
        cornerRadius: 0,
        dividerMargin: 5,
        dividerColor: Colors.grey,
        inactiveBgColor: Colors.transparent,
        customWidths: const [47, 51, 60, 58],
        labels: const ["Aus", "Fern", "Regio", "Lokal"],
        initialLabelIndex: 0,
        activeBgColor: const [
          Color.fromRGBO(55, 95, 226, 1.0),
          Color.fromRGBO(55, 95, 226, 1.0),
          Color.fromRGBO(55, 95, 226, 1.0),
          Color.fromRGBO(55, 95, 226, 1.0),
        ],
        onToggle: (value) {
        },
      ),
    ],
  ),
```

But maybe I'm just stupid and use this completely wrong 😅

**Comment by PramodJoshi** [comment]
@derklaro I added Flexible in 2.2.2 to give the toggle_switch needed space and avoid overflow issue. You'll need to adjust your `customWidths` or set a `minWidth`. I tried 'minWidth: 75.0' and here's what it looks like,

<img width="300" alt="Screenshot 2024-03-22 at 12 41 47 PM" src="https://github.com/PramodJoshi/toggle_switch/assets/17320471/1de8dd10-097b-41c8-a611-0c4cf6d11748">

Regarding center text, I think I found the issue. I'll try to fix that asap. Thanks for catching that.
